man/systemd-stub.xml

```diff
@@ -312,6 +312,16 @@
         <literal>13</literal>). This variable is set if a measurement was successfully completed, and remains
         unset otherwise.</para></listitem>
       </varlistentry>
+
+      <varlistentry>
+        <term><varname>StubSecureBootWorkaround</varname></term>
+
+        <listitem><para>A boolean indicating whether to try to work around secure boot restrictions. Setting
+        this to <literal>1</literal> (the default) may allow booting the embedded kernel image even if it is
+        not trusted by the firmware. Setting this to <literal>0</literal> will disable this and any attempts
+        to load an untrusted kernel will then fail. This option has no effect if the system is not in secure
+        boot mode. Note that the UKI itself has to be trusted by the firmware regardles of this option.</para></listitem>
```

Comment:
This text is not wrong, but it does not really explain things in a way that it useful to a user.
The term UKI does not appear otherwise in this text…
Also, let's not call the variable "workaround" — there might be another workaround in the future and such names get old quickly. Just call it after what it really does.

Maybe something along those lines:

StubDisableVerificationOfKernel

"A boolean indicating whether the boot stub should try to disable the secure boot check of the `<empasis>embedded</>` kernel image. When set to `<literal>1</literal>` (the default), the boot stub tells the firmware that the embedded kernel is trusted, which may allow the embedded kernel image to be booted even if it wouldn't be directly trusted by the firmware. This option has no effect if the system is not in secure boot mode. Note that the unified kernel image (UKI) containing the stub and the kernel must be trusted as a whole, so trust in the stub and the kernel is established before the boot stub is executed."

(I removed "firmware", because it isn't necessarily firmware that establishes the trust. It may be shim, and shim is not "firmware".)

Reply:
This doesn't seem to have been addressed…